tests/no_std_builds.rs

```diff
@@ -0,0 +1,11 @@
+use std::process::Command;
+
+#[test]
+fn build_no_std_macros() {
+    // Ensure that the macro crate and the macro-expanded code can work in a `no_std` environment.
+    let status = Command::new("cargo")
+        .args(["build", "-p", "test-no-std-macros"])
```

Comment:
FYI looks like building the default `x86_64-unknown-linux-gnu` target is enough here. The `test-no-std-macros` crate is marked as `#[no_std]`. Any use of `::std` in the macro-expanded code should fail the compilation, even with the `x86_64-unknown-linux-gnu` target. There is no need to build with the `x86_64-unknown-none` target here.

Reply:
Running cargo from inside the build script is very obscure and will fail in some environments. Let's add a CI job instead.

Reply:
Sure. The build test is moved to CI.